development/mimir-monolithic-mode/config/mimir.yaml

```diff
@@ -71,5 +71,14 @@ ruler_storage:
     secret_access_key: supersecret
     insecure: true
 
+query_scheduler:
+  # Comment to not use the query-scheduler and enqueue queries directly in the query-frontend.
+  service_discovery_mode: "ring"
+  ring:
+    kvstore:
+      store: consul
+      consul:
+        host: consul:8500
+
```

Comment:
> Do we expect someone to run it this way?

No, we don't. It was introduced to test manually it in monolithic mode too (e.g. the were some bugs in my first implementation). If you prefer, I can revert it.

Reply:
I'd prefer to remove it in future PRs, as this now doesn't test the _usual_ scenario.

OTOH, I wonder if it would make sense to have query-scheduler enabled by default in monolith in future versions, with memberlist ring?

Reply:
> I'd prefer to remove it in future PRs, as this now doesn't test the usual scenario.

See: https://github.com/grafana/mimir/pull/2987

> OTOH, I wonder if it would make sense to have query-scheduler enabled by default in monolith in future versions, with memberlist ring?

Yes, but the first pre-requisite would be switch to ring-based service discovery by default. We'll first get some confidence with the query-scheduler ring, then offer a migration path from DNS to ring, finally switch default to ring. Will be a long journey.